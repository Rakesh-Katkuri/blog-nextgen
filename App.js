import React, { useEffect, useState } from "react";
import { BrowserRouter, Route, Routes, Navigate } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import Home from "./home/Home";
import MyBlog from "./blogs/MyBlog";
import AddBlog from "./blogs/AddBlog";
import AuthProvider from "./authContext/AuthContext";
import Login from "./userAuthentication/LoginForm";
import Register from "./userAuthentication/RegisterForm";
import UpdatePostForm from "./blogs/UpdatePostFom";
import AllBlogs from "./blogs/AllBlogs";
import MyFavorites from "./myFavorites/MyFavorites";
import SearchedBlogs from "./searchBlogs/SearchedBlogs";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import BlogDetail from "./blogs/BlogDetail";
import Footer from "./footer/Footer";
import Navbar2 from "./home/Navbar2";
import ToastMessage from "./toastMessage/ToastMessages";
// import PrivateRoute from "./PrivateRoute";

const PrivateRoute = ({ children }) => {
  const userId = localStorage.getItem("userId");
 

  return userId ? children : <Navigate to="/login" />
  

  // useEffect(() => {
  //   // const userId = localStorage.getItem("userId");
  //   if (!userId) {
  //     // Show a toast notification for unauthorized access
  //     // toast.error("Authentication required. Please log in to access page.", {
  //     //   position: "top-center",
  //     //   autoClose: 5000,
  //     //   hideProgressBar: false,
  //     //   closeOnClick: true,
  //     //   pauseOnHover: true,
  //     //   draggable: true,
  //     // });
  //     return (
  //       <>
  //       <Navigate to="/login" />
  //       <div class="alert alert-danger">
  //   <strong>Danger!</strong> This alert box could indicate a dangerous or potentially negative action.
  // </div>
  //       </>
  //     )
  //   }
  // }, []);

  // if (!userId) {
  //   // Redirect to the login page if userId is not present
    
  //   return <Navigate to="/login" />
    
  
  // }

  // return <>{element}</>; // Only wrap the specific route element, not all Routes
};
const AlertMessage = ({show}) =>{

  toast.error("Authentication required. Please log in to access page.", {
        position: "top-center",
        autoClose: 3000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
      });

  return (
    <>
    {show ? <ToastContainer/> : null }
    
    </>
  )
}

// const Authentication = () =>{

//   const userId = localStorage.getItem("userId");
// if(!userId){
// setShowToast(true)
// }
  
// }

const App = () => {
  const userId = localStorage.getItem("userId");
  const [alert, setAlert] = useState(false)
  
  // Clear localStorage on app startup
  useEffect(() => {
    // const userId = localStorage.getItem("userId");
    
    if(userId){ 
      setAlert(true);
    }else{
      setAlert(false);
    }
    
  }, [userId]);

  return (
    <div className="conatiner">

<BrowserRouter>
      
      <AuthProvider>
      <Navbar2/>
      <AlertMessage show={alert}/>
        <Routes>
          <Route path="/" element={<AllBlogs />} />
          <Route path="/login" element={<Login />} />
          <Route path="/signup" element={<Register />} />

          {/* <Route path="/login" element={<PrivateRoute element={<Login />} />} /> */}
          <Route path="/" element={<PrivateRoute><AllBlogs /></PrivateRoute> }/>
          <Route
            path="/my-blogs"
            element={<PrivateRoute ><MyBlog /></PrivateRoute> } 
          />
          <Route
            path="/add-blog"
            element={<PrivateRoute><AddBlog /> </PrivateRoute>}
          />
          <Route
            path="/update/:postId"
            element={<PrivateRoute><UpdatePostForm /></PrivateRoute >}
          />
          <Route
            path="/my-favorite/blogs"
            element={<PrivateRoute><MyFavorites /></PrivateRoute>}
          />
          <Route
            path="/blog/:id"
            element={<PrivateRoute><BlogDetail /></PrivateRoute >}
          />
          <Route
            path="/search-results"
            element={<PrivateRoute><SearchedBlogs /></PrivateRoute> }
          /> 
          
        </Routes>
        <Footer/>
        {/* <ToastContainer/> */}
      </AuthProvider>
      
    </BrowserRouter>

     
      {/* <BrowserRouter>
      
        <AuthProvider>
        <Navbar2/>
          <Routes>
            <Route path="/" element={<AllBlogs />} />
            <Route path="/login" element={<Login />} />
            <Route path="/signup" element={<Register />} />

            <Route path="/login" element={<PrivateRoute element={<Login />} />} />
            <Route path="/" element={<PrivateRoute element={<AllBlogs />} />} />
            <Route
              path="/my-blogs"
              element={<PrivateRoute element={<MyBlog />} />}
            />
            <Route
              path="/add-blog"
              element={<PrivateRoute element={<AddBlog />} />}
            />
            <Route
              path="/update/:postId"
              element={<PrivateRoute element={<UpdatePostForm />} />}
            />
            <Route
              path="/my-favorite/blogs"
              element={<PrivateRoute element={<MyFavorites />} />}
            />
            <Route
              path="/blog/:id"
              element={<PrivateRoute element={<BlogDetail />} />}
            />
            <Route
              path="/search-results"
              element={<PrivateRoute element={<SearchedBlogs />} />}
            />
            
          </Routes>
          <Footer/>
        
        </AuthProvider>
        
      </BrowserRouter> */}
     
      
      </div>
  );
};
export default App;
//kilk