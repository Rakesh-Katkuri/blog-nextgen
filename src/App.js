import React, { useEffect } from "react";
import { BrowserRouter, Route, Routes, Navigate } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import Home from "./home/Home";
import MyBlog from "./blogs/MyBlog";
import AddBlog from "./blogs/AddBlog";
import AuthProvider from "./authContext/AuthContext";
import Login from "./userAuthentication/LoginForm";
import Register from "./userAuthentication/RegisterForm";
import UpdatePostForm from "./blogs/UpdatePostFom";
import AllBlogs from "./blogs/AllBlogs";
import MyFavorites from "./myFavorites/MyFavorites";
import SearchedBlogs from "./searchBlogs/SearchedBlogs";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import BlogDetail from "./blogs/BlogDetail";
// import PrivateRoute from "./PrivateRoute";

const PrivateRoute = ({ element }) => {
  const userId = localStorage.getItem("userId");

  useEffect(() => {
    const userId = localStorage.getItem("userId");
    if (!userId) {
      // Show a toast notification for unauthorized access
      toast.error("Authentication required. Please log in to access page.", {
        position: "top-center",
        autoClose: 5000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
      });
    }
  }, []);

  if (!userId) {
    // Redirect to the login page if userId is not present
    return <Navigate to="/login" />;
  }

  return <>{element}</>; // Only wrap the specific route element, not all Routes
};

const App = () => {
  // Clear localStorage on app startup
  useEffect(() => {
    const userId = localStorage.getItem("userId");
    if (!userId) {
      localStorage.removeItem("userId");
    }
  }, []);

  return (
    <>
      <BrowserRouter>
        <AuthProvider>
          <Routes>
            <Route path="/" element={<AllBlogs />} />
            <Route path="/login" element={<Login />} />
            <Route path="/signup" element={<Register />} />

            <Route path="/" element={<PrivateRoute element={<AllBlogs />} />} />
            <Route
              path="/my-blogs"
              element={<PrivateRoute element={<MyBlog />} />}
            />
            <Route
              path="/add-blog"
              element={<PrivateRoute element={<AddBlog />} />}
            />
            <Route
              path="/update/:postId"
              element={<PrivateRoute element={<UpdatePostForm />} />}
            />
            <Route
              path="/my-favorite/blogs"
              element={<PrivateRoute element={<MyFavorites />} />}
            />
            <Route
              path="/blog/:id"
              element={<PrivateRoute element={<BlogDetail />} />}
            />
            <Route
              path="/search-results"
              element={<PrivateRoute element={<SearchedBlogs />} />}
            />
          </Routes>
        </AuthProvider>
        <ToastContainer />
      </BrowserRouter>
    </>
  );
};
export default App;
//kilk
