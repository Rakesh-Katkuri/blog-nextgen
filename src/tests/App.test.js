import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { mount, shallow } from "enzyme";
import axios from "axios";
import { configure } from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";
import App from "../App";
configure({ adapter: new Adapter() });

jest.mock("axios");

describe("App component testing", () => {
  test("sample", () => {
    expect("nextgen").toBe("nextgen");
  });

  //Snapshot testing
  test("Snapshop testing for App component", () => {
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
  });

  //Render whole application
  test("Render whole App application", () => {
    const AppComponent = mount(<App />);
  });
  
});

// npm test -- --updateSnapshot --watch

//error get when use contextapi
//error
  //const { posts, getBlogs, updatePost, loggedIn } = useAuth(); and
  //const userId = localStorage.getItem("userId");
  //Cannot destructure property 'posts' of '(0 , _AuthContext.useAuth)(...)' as it is undefined.
