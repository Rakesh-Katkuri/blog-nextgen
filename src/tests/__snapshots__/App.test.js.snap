// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing Snapshop testing for App component 1`] = `
<div
  className="login template d-flex justify-content-center align-items-center vh-100 bg-dark-subtle "
>
  <div
    className="form_container 50-w p-5 rounded bg-white"
  >
    <form>
      <h2
        className="text-center mb-4"
      >
        User SignIn
      </h2>
      <div
        className="mb-2"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="em"
          name="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
        />
      </div>
      <div
        className="mb-4"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="pw"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div
        className="d-grid"
      >
        <button
          className="btn btn-primary btn-block"
          onClick={[Function]}
          type="submit"
        >
          Sign In
        </button>
      </div>
      <p
        className="text-end mt-2"
      >
        Forgot 
        <a
          href=""
        >
          Password?
        </a>
         
        <a
          className="ms-2"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
         
      </p>
    </form>
  </div>
</div>
`;
